src/Common/src/Interop/Windows/HttpApi/Interop.HttpApi.cs

```diff
@@ -0,0 +1,1017 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Win32.SafeHandles;
+using System;
+using System.Collections.Generic;
+using System.Net;
+using System.Net.Sockets;
+using System.Runtime.InteropServices;
+using System.Security.Authentication.ExtendedProtection;
+using System.Threading;
+
+internal static partial class Interop
+{
+    internal unsafe static partial class HttpApi
+    {
+        internal static readonly HTTPAPI_VERSION Version = new HTTPAPI_VERSION() { HttpApiMajorVersion = 2, HttpApiMinorVersion = 0 };
+
+        internal static readonly bool Supported = InitHttpApi(Version);
+
+        private static bool InitHttpApi(HTTPAPI_VERSION version)
+        {
+            uint statusCode = HttpInitialize(version, (uint)HTTP_FLAGS.HTTP_INITIALIZE_SERVER, null);
```

Comment:
@Priya91 To clarify: this is good as written and no more changes are required. CoreFX will only run on Win7 and above which support the above initialization code.